react-js/react-components-and-hierarchies.md

```diff
@@ -0,0 +1,479 @@
+      <Button color="link">link</Button>
+    </div>
+  );
+}
+
+export default App;
+```
+##### Reactstrap Navbar in React Project
+The reactstrap navbar can be used to create responsive navigation bar in React app. It includes many other small components such as Collapse, Navbar, NavbarToggler, Nav, NavItem, NavLink, DropDownMenu, etc.
+```
+import React from 'react';
+import { Fragment } from 'react';
+import {
+  Collapse,
+  Navbar,
+  NavbarToggler,
+  NavbarBrand,
+  Nav,
+  NavItem,
+  NavLink,
+  UncontrolledDropdown,
+  DropdownToggle,
+  DropdownMenu,
+  DropdownItem
+} from 'reactstrap';
+
+export default class App extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.toggle = this.toggle.bind(this);
+    this.state = {
+      isOpen: false
+    };
+  }
+  toggle() {
+    this.setState({
+      isOpen: !this.state.isOpen
+    });
+  }
+
+  render() {
+    return <Fragment>
+      <Navbar color="light" light expand="md">
+        <NavbarBrand href="/">reactstrap</NavbarBrand>
+        <NavbarToggler onClick={this.toggle} />
+        <Collapse isOpen={this.state.isOpen} navbar>
+          <Nav className="ml-auto" navbar>
+            <NavItem>
+              <NavLink href="/components/">Components</NavLink>
+            </NavItem>
+            <NavItem>
+              <NavLink href="https://github.com/reactstrap/reactstrap">GitHub</NavLink>
+            </NavItem>
+            <UncontrolledDropdown nav inNavbar>
+              <DropdownToggle nav caret>
+                Options
+                </DropdownToggle>
+              <DropdownMenu right>
+                <DropdownItem>
+                  Option 1
+                  </DropdownItem>
+                <DropdownItem>
+                  Option 2
+                  </DropdownItem>
+                <DropdownItem divider />
+                <DropdownItem>
+                  Reset
+                  </DropdownItem>
+              </DropdownMenu>
+            </UncontrolledDropdown>
+          </Nav>
+        </Collapse>
+      </Navbar>
+    </Fragment>
+  }
+}
+```
+
+##### Implementing Reactstrap Modal in React
+The reactstrap Modal component creates a Bootstrap Modal with a header, a body, and a footer.
+```
+<Modal isOpen={this.state.show} toggle={this.toggleModal} >
+
+    <ModalHeader toggle={this.toggle}>
+        Modal title
+    </ModalHeader>
+
+    <ModalBody>
+        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
+    </ModalBody>
+
+    <ModalFooter>
+        <Button color="primary" onClick={this.toggle}>Do Something</Button>{' '}
+        <Button color="secondary" onClick={this.toggle}>Cancel</Button>
+     </ModalFooter>
+</Modal>
+```
+
+<br/><br/>
+
+### Guided Practice
+> #### Let's create a to do list application using react
```

Comment:
The guided practice is meant to cover the same material as the above, not go into deeper complexity.